server/controllers/joinLeaveCancelController.js

```diff
@@ -135,30 +146,48 @@ const handleCancelEvent = async (req, res, next) => {
   try {
     const eventId = req.body.eventId;
     if (!eventId) {
-      res.status(400).json({
-        status: 400,
-        message: "Missing information from event inside handleCancelEvent.",
-      });
+      let err = new NotFoundError("Missing information from event");
+      next(err);
     }
     let getEvent = await getEventById(eventId); //get the event for participantId.
+    if (!getEvent) {
+      let err = new NotFoundError("Unable to get event in handleCancelEvent.");
+      next(err);
+    }
     let deletedRoom = await deleteRoom(eventId); //for chat system.
-    let deletedEvent = await deleteEvent(eventId); //delet
+    let deletedEvent = await deleteEvent(eventId); //delete event.
+    console.log(deletedRoom, "DELETEDROOM");
+    if (!deletedEvent || !deletedRoom) {
+      let err = new ConflictError(
```

Comment:
A conflict error is a bit weird here, there is probably a better error type for this